src/main/scala/com/modelfabric/sparql/stream/client/Builder.scala

```diff
@@ -138,6 +139,7 @@ object Builder {
       import GraphDSL.Implicits._
       import endpoint._
 
+      // JC: why all stages in sparqlQueryFlow are async, but not here? do we need to make all of them async?
```

Comment:
Ah, I meant to make this async as well. BTW, AFAIK, without making them async the materializer would make an attempt to compress them into as few flow stages as possible. Making them async ensures these are materialised as separate actors. We can play around with this when we do performance tuning.